File: test/cpp/case31_slice_basic.cpp
// case30: スライス構文 a[b:c] の基本テスト（stepなし）。


#include "cpp_module/py_runtime.h"

// case30: スライス構文 a[b:c] の基本テスト（stepなし）。

void __pytra_main() {
    list<int64> nums = list<int64>{10, 20, 30, 40, 50};
    str text = "abcdef";
    
    list<int64> mid_nums = py_slice(nums, 1, 4);
    str mid_text = py_slice(text, 2, 5);
    
    py_print(mid_nums[0]);
    py_print(mid_nums[2]);
    py_print(mid_text);
}

int main() {
    __pytra_main();
    return 0;
}

File: test/py/case31_slice_basic.py
# case30: スライス構文 a[b:c] の基本テスト（stepなし）。

def main() -> None:
    nums: list[int] = [10, 20, 30, 40, 50]
    text: str = "abcdef"

    mid_nums: list[int] = nums[1:4]
    mid_text: str = text[2:5]

    print(mid_nums[0])
    print(mid_nums[2])
    print(mid_text)


if __name__ == "__main__":
    main()
